### PR TITLE
fix(rest-api-model-server): fix compability

### PR DESCRIPTION
**Pull request by odzhychko**
See comments in the changed file for details.

The problem is, that we want to use the platform release and that the platform release has broken (for some cases, like our Quarkus case) Ktor version.

The issue arises because of an older Quarkus version, which needs an older Kotlin version.

### CODE REVIEW

**Comment by languitar** [comment]
We should bump to Quarkus 3 soon in any case.

File: rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Test
+import org.modelix.model.client.ReplicatedRepository
+import javax.inject.Inject
+
+@QuarkusTest
+class ReplicatedRepositoryInstantiationTest {
+    @Inject
+    var replicatedRepository: ReplicatedRepository? = null
```
Comment (languitar) [inline rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt:13]: We've usuallu used a lateinit var for these cases. Constructor injection could also work here

**Review by languitar — APPROVED** [review]
Just one idea for a more canonical test structure. Nice corner case catch and explanations.

**Review by nkoester — APPROVED** [review]
Fixes the previous issues and the quarkus API  works with this fix

File: rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt (hunk 1 above)
Comment (odzhychko) [reply rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt]: Done.
Used `lateinit`.
Constructor injection does not work. See. https://github.com/quarkusio/quarkus/issues/34330

File: rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        // This tests, whether `replicatedRepository` can be correctly injected and instantiated by Quarkus.
+        // The actual instantiation is happening when we call something on `replicatedRepository`.
+        // Therefore, we call `.branch`.
+        // The asserted exception message indicates that the `replicatedRepository` was injected correctly.
+        // The exception message "java.lang.NoClassDefFoundError: kotlin/enums/EnumEntriesKt", for example,
+        // would indicate that something is set up wrong and the instantiation is failing.
+        // In the past, we had such issues with the wrong version of Kotlin and Ktor.
+        assertThrows(
+            RuntimeException::class.java,
+            { replicatedRepository?.branch },
```
Comment (languitar) [inline rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt:26]:
```suggestion
            { replicatedRepository.branch },
```

It's not nullable anymore with lateinit

Comment (odzhychko) [reply rest-api-model-server/src/test/kotlin/ReplicatedRepositoryInstantiationTest.kt]: Done ^^'

I relied too long on linting on commits

**Review by nkoester — APPROVED** [review]
